app/features/user/user-repository.ts

```diff
@@ -169,7 +169,14 @@ export class UserRepository {
     }
 
     const { accounts, ...upsertedUser } = data;
-    return this.toUser(upsertedUser);
+    return {
+      user: this.toUser(upsertedUser),
+      accounts: await Promise.all(
```

Comment:
I did this and then preloaded cache with this data because the data was already there and now we don't need to suspend to wait for additional request to fetch the accounts. this shaved of some milliseconds as well